hibernate-reactive-core/src/main/java/org/hibernate/reactive/stage/impl/StageSessionFactoryImpl.java

```diff
@@ -133,10 +134,12 @@ private ReactiveConnection proxyConnection(String tenantId) {
 		return newSession().thenCompose(
 				session -> {
 					Vertx.currentContext().putLocal(id, session);
-					return work.apply(session).whenComplete( (r, e) -> {
-						Vertx.currentContext().removeLocal(id);
-						session.close();
-					} );
+					return work.apply(session)
+							.handle(this::handler)
+							.thenCompose( handler -> {
+						        Vertx.currentContext().removeLocal(id);
+								return session.close().thenApply(handler);
+							} );
```

Comment:
@DavideD @Sanne note that this fixes a really bad bug on main where `Stage` sessions opened using `withSession()` _still_ weren't getting closed after the migration to Vert.x 4.

Reply:
sorry unable to review it all right now, but if there's some urgent bugfix in here could you extract it for immediate integration?

We're very close to deadlines for Quarkus 2 : we'll need another tag of HR tomorrow - as soon as I manage to get an ORM tag released to match it.

Reply:
Well, TBH I'm pretty happy with this work now, so if I add a couple more tests and update the docs, I think it would be fine to merge it.

Reply:
Ok!